src/colorlibrary.cpp

```diff
@@ -1,18 +1,43 @@
 #include "colorlibrary.h"
 
+const static QVector<QColor> DefaultColors =
+{
+    QColor("#f2ec99"),
+    QColor("#bfe4f3"),
+    QColor("#bfe968"),
+    QColor("#c6caf8"),
+    QColor("#eec757"),
+    QColor("#9994f1"),
+    QColor("#edb0e8"),
+    QColor("#cebff3"),
+    QColor("#a5f0c6"),
+    QColor("#f5b0b4")
+};
+
 ColorLibrary::ColorLibrary()
 {
     m_colorLibrary = new QQueue<QColor>();
-    m_colorLibrary->enqueue(QColor("#f2ec99"));
-    m_colorLibrary->enqueue(QColor("#bfe4f3"));
-    m_colorLibrary->enqueue(QColor("#bfe968"));
-    m_colorLibrary->enqueue(QColor("#c6caf8"));
-    m_colorLibrary->enqueue(QColor("#eec757"));
-    m_colorLibrary->enqueue(QColor("#9994f1"));
-    m_colorLibrary->enqueue(QColor("#edb0e8"));
-    m_colorLibrary->enqueue(QColor("#cebff3"));
-    m_colorLibrary->enqueue(QColor("#a5f0c6"));
-    m_colorLibrary->enqueue(QColor("#f5b0b4"));
+    AddColors(DefaultColors);
+}
+
+ColorLibrary::ColorLibrary(QVector<QColor> usedColors)
+{
+    m_colorLibrary = new QQueue<QColor>();
```

Comment:
m_colorLibrary can be changed to class instance var instead of a pointer. Then no need to init in both constructors and deal with the missing deletes.